GettingStarted.md

```diff
@@ -1,12 +1,176 @@
+```
+
+* Non forms classes can use the same interface using
+```
+    ILocalytics localytics = LocalyticsSDK.SharedInstance;
+``` 
+
+## API Summary
+
+API's follow c# naming conventions and are translation of the API on the native platform.
+
+### Common APIs
+
+| API                    | Description  |
+|------------------------|------|
+| OpenSession  | Opens a session. Multiple calls are coallesed.  |
+| CloseSession | Open Sessions are marked with a pending close. Sessions are extended if there is localytics activity before expiry of Session timer |
+| Upload| uploads any data stored on the device by the localytics SDK. Essential to do this early, to ensure upload completes before app is suspended.|
+| PauseDataUploading | all data upload is deferred until it is resumed. Calls to the upload API dont perform any action. When data upload is resumed, all locally stored data is immediately uploaded. |
+| TagEvent | Tag an event |
+| TagPurchased | A standard event to tag a single item purchase event (after the action has occurred) |
+| TagAddedToCart | A standard event to tag the addition of a single item to a cart (after the action has occurred) |
+| TagStartedCheckout | A standard event to tag the start of the checkout process (after the action has occurred) |
+| TagCompletedCheckout | A standard event to tag the conclusions of the checkout process (after the action has occurred) |
+| TagContentViewed | A standard event to tag the viewing of content (after the action has occurred) |
+| TagSearched | A standard event to tag a search event (after the action has occurred) |
+| TagShared | A standard event to tag a share event (after the action has occurred) |
+| TagContentRated | A standard event to tag the rating of content (after the action has occurred) |
+| TagCustomerRegistered | A standard event to tag the registration of a user (after the action has occurred) |
+| TagCustomerLoggedIn | A standard event to tag the logging in of a user (after the action has occurred) |
+| TagCustomerLoggedOut | A standard event to tag the logging out of a user (after the action has occurred) |
+| TagInvited | A standard event to tag the invitation of a user (after the action has occured) |
+| TagScreen | Allows tagging the flow of screens encountered during the session. |
+| SetCustomDimension | Sets the value of custom dimension which is user defined data. Customer sensitive data should be hashed or encrypted |
+| GetCustomDimension |Gets the custom value for a given dimension. Must not be called from the main thread. |
+| SetIdentifier | Sets the value of a custom identifier |
+| GetIdentifier | Gets the identifier value for a given identifier. Must not be called form the main thread. |
+| CustomerId | property representing a customer Id. Recommended to use SetCustomerId. privacy sensitive data should be hashed or encyrpted |
+| SetCustomerId | set customer Id and privacy status automically.|
```

Comment:
Looking at this method makes me think we should come up with some convention to indicate what the expected parameter types are for all of these methods.